src/Compilers/CSharp/Portable/Symbols/Source/CustomModifierUtils.cs

```diff
@@ -35,8 +35,7 @@ internal static class CustomModifierUtils
             // have already been compared.
             MethodSymbol constructedSourceMethod = sourceMethod.ConstructIfGeneric(destinationMethod.TypeArguments);
 
-            customModifiers = CustomModifiersTuple.Create(constructedSourceMethod.ReturnTypeCustomModifiers,
-                                                          destinationMethod.ReturnsByRef ? constructedSourceMethod.RefCustomModifiers : ImmutableArray<CustomModifier>.Empty);
+            customModifiers = CustomModifiersTuple.Create(constructedSourceMethod.ReturnTypeCustomModifiers, constructedSourceMethod.RefCustomModifiers);
```

Comment:
Will adjust to logic to only copy on ref and ref-readonly. Will add a test to make sure we produce an error when a non-ref method is trying to override a ref-method, and that the symbol we create for the child method has no modifiers.

---
In reply to: [118421550](https://github.com/dotnet/roslyn/pull/19658#discussion_r118421550) [](ancestors = 118421550)